types/index.d.ts

```diff
@@ -759,11 +770,18 @@ declare namespace math {
      * b
      */
     lusolve(
-      A: Matrix | MathArray | number,
+      A: Matrix,
       b: Matrix | MathArray,
       order?: number,
       threshold?: number
-    ): Matrix | MathArray
+    ): Matrix
+
+    lusolve(
+      A: MathArray,
+      b: Matrix | MathArray,
```

Comment:
The behavior is: when one of the two inputs is a Matrix, the output will be a Matrix. So I think this type is not accurate, it can return a Matrix when `b` is a Matrix

Reply:
Just tried it, it seems like like the output is only ever a matrix when `A` is a matrix. Seems to return an array if A is an array and b is a matrix

Reply:
Ahh, you're right, nevermind. This is not entirely following the "array in -> array out, matrix in -> matrix out" `lusolve`, so it's a bit inconsistent, but let's not address that here.